src/main/java/org/veupathdb/service/eda/compute/plugins/correlation/CorrelationPlugin.java

```diff
@@ -0,0 +1,114 @@
+      idColumns.add(ancestor.getIdColumnDef());
+    }
+
+    HashMap<String, InputStream> dataStream = new HashMap<>();
+    dataStream.put(INPUT_DATA, getWorkspace().openStream(INPUT_DATA));
+    
+    RServe.useRConnectionWithRemoteFiles(dataStream, connection -> {
+      connection.voidEval("print('starting correlation computation')");
+
+      // Read in the abundance data
+      List<VariableSpec> computeInputVars = ListBuilder.asList(computeEntityIdVarSpec);
+      computeInputVars.addAll(util.getChildrenVariables(computeConfig.getCollectionVariable()));
+      computeInputVars.addAll(idColumns);
+      connection.voidEval(util.getVoidEvalFreadCommand(INPUT_DATA, computeInputVars));
+      connection.voidEval("abundanceData <- " + INPUT_DATA); // Renaming here so we can go get the sampleMetadata later
+
+      // Read in the sample metadata
+      // TO DO get only the continuous sample metadata!!
+      // List<VariableSpec> sampleMetadataVars = ListBuilder.asList(comparisonVariableSpec);
+      // sampleMetadataVars.add(computeEntityIdVarSpec);
+      // connection.voidEval(util.getVoidEvalFreadCommand(INPUT_DATA, sampleMetadataVars));
+      // connection.voidEval("sampleMetadata <- " + INPUT_DATA);
+
+
+      // Turn the list of id columns into an array of strings for R
+      List<String> dotNotatedIdColumns = idColumns.stream().map(VariableDef::toDotNotation).toList();
+      String dotNotatedIdColumnsString = "c(";
+      boolean first = true;
+      for (String idCol : dotNotatedIdColumns) {
+        if (first) {
+          first = false;
+          dotNotatedIdColumnsString = dotNotatedIdColumnsString + singleQuote(idCol);
+        } else {
+          dotNotatedIdColumnsString = dotNotatedIdColumnsString + "," + singleQuote(idCol);
+        }
+      }
+      dotNotatedIdColumnsString = dotNotatedIdColumnsString + ")";
+
+
+      // Set up input assay data.
+      // TEMP until we have the continuous sample metadata, we're letting the abundance data sub for continuous sample metadata
+      connection.voidEval("inputData <- microbiomeComputations::AbundanceData(data=abundanceData" + 
+                                                                          ", sampleMetadata=abundanceData" +
+                                                                          ", recordIdColumn=" + singleQuote(computeEntityIdColName) +
+                                                                          ", ancestorIdColumns=as.character(" + dotNotatedIdColumnsString + ")" +
+                                                                          ", imputeZero=TRUE)");
+
+
+      connection.voidEval("computeResult <- microbiomeComputations::correlation(data=inputData" +
+                                                          ", method=" + singleQuote(method) +
+                                                          ", verbose=TRUE)");
+
+
+      String statsCmd = "writeStatistics(computeResult, NULL, TRUE)";
```

Comment:
related to a comment i made in the microbiomeComputations PR.. if we would like more descriptive names than 'var1' and 'var2' in the response then we may need to introduce some type of computed variable metadata here.. its a bit different thought than how we used it for the beta div app since these 'computed vars' cant go through the merge service though..